aiowebostv/webos_client.py

```diff
@@ -613,28 +619,33 @@ def _parse_response(self, response: dict[str, Any]) -> dict[str, Any]:
 
         return payload
 
-    async def create_subscription_handler(self, uid: int, callback: Callable) -> None:
-        """Create a subscription handler for a given uid.
+    async def create_subscription_handler(
+        self, uri: str, uid: int, callback: Callable
+    ) -> None:
+        """Create a subscription handler for a given uri & uid.
 
         Create a queue to store the messages, a task to handle the messages
         and a future to signal first subscription update processed.
         """
+        self._subscriptions[uri] = uid
         self.futures[uid] = future = self._loop.create_future()
         queue: Queue[dict[str, Any]] = asyncio.Queue()
         self.callback_queues[uid] = queue
         self.callback_tasks[uid] = asyncio.create_task(
             self.callback_handler(queue, callback, future)
         )
 
-    async def delete_subscription_handler(self, uid: int) -> None:
-        """Delete a subscription handler for a given uid."""
+    async def delete_subscription_handler(self, uri: str) -> None:
+        """Delete a subscription handler for a given uri."""
+        uid = self._subscriptions[uri]
         task = self.callback_tasks.pop(uid)
         if not task.done():
             task.cancel()
         while not task.done():
             with suppress(asyncio.CancelledError):
                 await task
         del self.callback_queues[uid]
+        del self._subscriptions[uri]
```

Comment:
Is there a difference between `self._subscriptions.pop(..)` and `del`?

Reply:
Only that del doesn't return the item and a neglectable performance improvement